packages/backend/src/graphql/resolvers/invoice.test.ts

```diff
@@ -0,0 +1,223 @@
+import { gql } from 'apollo-server-koa'
+import Knex from 'knex'
+
+import { createTestApp, TestContainer } from '../../tests/app'
+import { IocContract } from '@adonisjs/fold'
+import { AppServices } from '../../app'
+import { initIocContainer } from '../..'
+import { Config } from '../../config/app'
+import { User as UserModel } from '../../user/model'
+import { UserService } from '../../user/service'
+import { truncateTables } from '../../tests/tableManager'
+import { Invoice } from '../../invoice/model'
+import { InvoiceService } from '../../invoice/service'
+import { User } from '../generated/graphql'
+
+describe('Invoice Resolver', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let invoiceService: InvoiceService
+  let userService: UserService
+  let knex: Knex
+  let invoices: Invoice[]
+  let user: UserModel
+
+  beforeAll(
+    async (): Promise<void> => {
+      deps = await initIocContainer(Config)
+      appContainer = await createTestApp(deps)
+      knex = await deps.use('knex')
+      invoiceService = await deps.use('invoiceService')
+      userService = await deps.use('userService')
+      user = await userService.create()
+      invoices = []
+      for (let i = 0; i < 50; i++) {
+        invoices.push(await invoiceService.create(user.id))
+      }
+    }
+  )
+
+  afterAll(
+    async (): Promise<void> => {
+      await appContainer.apolloClient.stop()
+      await appContainer.shutdown()
+      await truncateTables(knex)
+    }
+  )
+
+  describe("User's invoices", (): void => {
```

Comment:
Are tests needed for the following situations:
* no invoices, but invoices requested
* request invoices past the 50th? (I see the after = 44, first = 10.  What about after = 50, first = 10?
* negative index for pagination? (after = -1, first = -1)

Reply:
> no invoices, but invoices requested

:+1: 

> request invoices past the 50th? (I see the after = 44, first = 10. What about after = 50, first = 10?

Could you clarify why this would be useful? There are only 50 dummy invoices created for testing. Also, `after` is not an index it's a cursor.

> negative index for pagination? (after = -1, first = -1)

This is tested in the service, along with other pagination mechanics and limits.

Reply:
> request invoices past the 50th? (I see the after = 44, first = 10. What about after = 50, first = 10?

If it's already tested, no worries.  I was wondering what happens on the edge cases where we go beyond the available invoices.  Trigger any errors that might occur.  If it's already tested, no worries.